rules/coding-style/src/Rector/Ternary/TernaryConditionVariableAssignmentRector.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\Ternary;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\Ternary;
+use PhpParser\Node\Expr\Variable;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\CodingStyle\Tests\Rector\Ternary\TernaryConditionVariableAssignmentRector\TernaryConditionVariableAssignmentRectorTest
+ */
+final class TernaryConditionVariableAssignmentRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Assign outcome of ternary condition to variable, where applicable', [
+            new CodeSample(
+                <<<'PHP'
+function ternary()
+{
+    true ? $a = 1 : $a = 0;
```

Comment:
The `true` value is a bit confusing here, as it could be any expr (falseable).

Adding a function arg `$value` and using it would make it more clear.